CPP-220 S3135 "sizeof" should not be called on pointer parameters (#1030)
SonarSource/rspec: rules/S3135/cfamily/rule.adoc

`sizeof` returns the size in bytes of a type. One common usage pattern, especially in C, is to use `sizeof` to determine the size of an array. However, arrays decay to pointers when passed as arguments to a function, and if `sizeof` is applied to such an argument, it will return the size of the pointer, not of the array. A similar issue happens when the array is used in an arithmetic operation.

> This rule raises issues when `sizeof` is used to compute the array size of a pointer passed as a function argument or used in an arithmetic operation.

Reviewer: I find the last part of the sentence a bit misleading: the problem is not when `sizeof` itself is part of an arithmetic operation but when it takes an arithmetic operation involving an array as an argument. 
What about something like:
```suggestion
This rule raises issues when:
* `sizeof` is used to compute the array size of a pointer passed as a function argument
* `sizeof` is called with an arithmetic operation involving an array
```

`sizeof` returns the size in bytes of a type. One common usage pattern, especially in C, is to use `sizeof` to determine the size of an array. However, arrays decay to pointers when passed as arguments to a function, and if `sizeof` is applied to such an argument, it will return the size of the pointer, not of the array. A similar issue happens when the array is used in an arithmetic operation.

This rule raises issues when:

> * `sizeof` is used to compute the array size of a pointer passed as a function argument.
> * `sizeof` is called on the result of an arithmetic operation involving an array.

Reviewer: ```suggestion
* `sizeof` is used to compute the array size of a pointer passed as a function argument
* `sizeof` is called on the result of an arithmetic operation involving an array
```

Author: I don't like these dots either, but I checked in the style guides that we use, and there is very little leeway for that: We cannot omit them because they form a full sentence...:
https://www.codot.gov/business/grants/safetygrants/assets/APStyleGuideCheatSheet.pdf
https://www.instructionalsolutions.com/blog/bulleted-list-punctuation

Reviewer: [Styling guide](https://github.com/SonarSource/rspec/blob/master/docs/styling_guide.adoc) is not precise on this subject.
Having a quick look at how rpsecs were currently written, we chose to keep the dots.
